specification/iotcentral/data-plane/Microsoft.IoTCentral/preview/1.2-preview/iotcentral.json

```diff
@@ -0,0 +1,5641 @@
+      "x-ms-parameter-location": "method",
+      "type": "string",
+      "required": true
+    },
+    "DeviceId": {
+      "in": "path",
+      "name": "deviceId",
+      "description": "Unique ID of the device.",
+      "x-ms-parameter-location": "method",
+      "type": "string",
+      "required": true
+    },
+    "DeviceTemplateId": {
+      "in": "path",
+      "name": "deviceTemplateId",
+      "description": "Unique [Digital Twin Model Identifier](https://github.com/Azure/opendigitaltwins-dtdl/blob/master/DTDL/v2/dtdlv2.md#digital-twin-model-identifier) of the device template.",
+      "x-ms-parameter-location": "method",
+      "type": "string",
+      "required": true
+    },
+    "ExportId": {
+      "in": "path",
+      "name": "exportId",
+      "description": "Unique ID for the export.",
+      "x-ms-parameter-location": "method",
+      "type": "string",
+      "required": true
+    },
+    "Filter": {
+      "in": "query",
+      "name": "$filter",
+      "description": "Filter string for results.",
+      "x-ms-parameter-location": "method",
+      "type": "string",
+      "required": false
+    },
+    "JobId": {
+      "in": "path",
+      "name": "jobId",
+      "description": "Unique ID of the job.",
+      "x-ms-parameter-location": "method",
+      "type": "string",
+      "required": true
+    },
+    "ModuleName": {
+      "in": "path",
+      "name": "moduleName",
+      "description": "Name of the device module.",
+      "x-ms-parameter-location": "method",
+      "type": "string",
+      "required": true
+    },
+    "OrderBy": {
+      "in": "query",
+      "name": "$orderby",
+      "description": "The order by string for results.",
+      "x-ms-parameter-location": "method",
+      "type": "string",
+      "required": false
+    },
+    "RelationshipId": {
+      "in": "path",
+      "name": "relationshipId",
+      "description": "Unique ID of a relationship between devices.",
+      "x-ms-parameter-location": "method",
+      "type": "string",
+      "required": true
+    },
+    "Subdomain": {
+      "in": "path",
+      "name": "subdomain",
+      "description": "The application subdomain.",
+      "x-ms-parameter-location": "client",
+      "type": "string",
+      "required": true
+    },
+    "TelemetryName": {
+      "in": "path",
+      "name": "telemetryName",
+      "description": "Name of this device telemetry.",
+      "x-ms-parameter-location": "method",
+      "type": "string",
+      "required": true
+    },
+    "Top": {
+      "in": "query",
+      "name": "$top",
+      "description": "Page size of returned results.",
+      "x-ms-parameter-location": "method",
+      "type": "integer",
+      "required": false
+    },
+    "Version": {
+      "in": "query",
+      "name": "api-version",
+      "description": "The version of the API being called.",
+      "x-ms-parameter-location": "client",
+      "type": "string",
+      "enum": [
+        "1.2-preview"
+      ],
```

Comment:
The `api-version` parameter should not be an enum.  Please use `x-ms-client-default` if your intent is to set this parameter to the latest version if not set by the user.

Reply:
This does not means setup an default,  but restrict the version to 1.2-preview, checked several specs, seems this is like a pattern to use it.
https://github.com/Azure/azure-rest-api-specs/blob/e7d2d8c48cf6f8f63de7e252c467930449b5fd88/specification/cosmos-db/data-plane/Microsoft.Tables/preview/2019-02-02/table.json#L1749

https://github.com/Azure/azure-rest-api-specs/blob/e7d2d8c48cf6f8f63de7e252c467930449b5fd88/specification/digitaltwins/data-plane/Microsoft.DigitalTwins/stable/2020-10-31/digitaltwins.json#L1555

Some are not specified, what's the recommendation here, should I just remove the enum?

Reply:
I recommend you just drop the enum, because if you leave it and want to change the single "known value" in the next version then it will be flagged as a breaking change (removing an enum value) and you'll have to come to the breaking change review board and explain why you are making a breaking change. Better to just avoid the hassle and leave the enum off.

I will make the same recommendation to Azure Tables and Digital Twins the next time they update their API and try to change the enum value for api-version.

Reply:
Thanks @mikekistler , will remove it.